avocado/utils/process.py

```diff
@@ -135,6 +136,57 @@ def safe_kill(pid, signal):  # pylint: disable=W0621
         return False
 
 
+def get_parent_pid(pid):
+    """
+    Returns the parent PID for the given process
+
+    TODO: this is currently Linux specific, and needs to implement
+    similar features for other platforms.
+
+    :param pid: The PID of child process
+    :returns: The parent PID
+    :rtype: int
+    """
+    with open('/proc/%d/stat' % pid) as proc_stat:
+        parent_pid = proc_stat.read().split(b' ')[3]
+        return int(parent_pid)
+
+
+def _get_pid_from_proc_pid_stat(proc_path):
+    match = re.match(r'\/proc\/([0-9]+)\/.*', proc_path)
+    if match is not None:
+        return int(match.group(1))
+
+
+def get_children_pids(parent_pid, recursive=False):
+    """
+    Returns the children PIDs for the given process
+
+    TODO: this is currently Linux specific, and needs to implement
+    similar features for other platforms.
+
+    :param parent_pid: The PID of parent child process
+    :returns: The PIDs for the children processes
+    :rtype: list of int
+    """
+    proc_stats = glob.glob('/proc/[123456789]*/stat')
+    children = []
+    for proc_stat in proc_stats:
+        try:
+            with open(proc_stat) as proc_stat_fp:
+                this_parent_pid = int(proc_stat_fp.read().split(b' ')[3])
+        except IOError:
```

Comment:
But, when this fails on Python 2, an `IOError` is raised, so that must be caught.

And, on Python 3.3+:

```
>>> open('/dev/foo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: '/dev/foo'
>>> issubclass(FileNotFoundError, OSError)
True
>>> issubclass(FileNotFoundError, IOError)
True
>>> IOError == OSError
True
```